src/apps/replication/lib/mutation_log.cpp

```diff
@@ -1126,10 +1159,11 @@ int mutation_log::garbage_collection(replica_log_info_map& durable_decrees)
                 }
 
                 global_partition_id gpid = kv.first;
-                decree durable_decree = kv.second.max_decree;
+                decree gc_max_decree = kv.second.max_decree;
                 int64_t valid_start_offset = kv.second.valid_start_offset;
 
-                bool delete_ok_for_this_replica;
+                bool delete_ok_for_this_replica = false;
+                bool kickout_this_replica = false;
```

Comment:
Yes, the refactor does not change the logic of the code.

My consider is, though kickout_this_replica here is sync with delete_ok_for_this_replica, but it is not necessary. For example, when max_decree map is empty, delete_ok_for_this_replica should be true, but kickout_this_replica may be false.

I considered the case when a log file is broken and useless, then the max_decree map is empty. But currently we assume that this case will not happen, and assert it at line 1172. So I did not handle this case in the code. But maybe someday we need to handle it....